src/SchemaRegistry.ts

```diff
@@ -149,6 +145,10 @@ export default class SchemaRegistry {
       },
     })
 
+    if (compatibility) {
```

Comment:
The behavior has changed a bit now. The intention of the code before was to only set the compatibility if there was no compatibility config already set. Now, we're setting the compatibility every time you call `register`. This should only happen if we get a 404 when getting the config.

Reply:
Ah, right! I added an extra guard to the `updateConfig` call. WDYT of that approach?